spec/lib/hubspot/company_properties_spec.rb

```diff
@@ -14,158 +14,325 @@
+              fieldType: "text"
+            })
+            Hubspot::CompanyProperties.create!({
+              name: "fax_number",
+              label: "Fax Number",
+              description: "The company fax number.",
+              groupName: group_b,
+              type: "string",
+              fieldType: "text"
+            })
+
+            response = Hubspot::CompanyProperties.all({}, { exclude: [group_a] })
+
+            assert_hubspot_api_request(:get, "/companies/v2/properties")
+
+            group_names = response.map { |property| property["groupName"] }
+            expect(group_names).not_to include(group_a)
+            expect(group_names).to include(group_b)
+
+            Hubspot::CompanyProperties.delete!("fax_number")
+            Hubspot::CompanyProperties.delete!("instagram_handle")
+          end
         end
       end
     end
 
-    let(:params) { {
-      'name'                          => 'my_new_property',
-      'label'                         => 'This is my new property',
-      'description'                   => 'What kind of x would you like?',
-      'groupName'                     => 'companyinformation',
-      'type'                          => 'string',
-      'fieldType'                     => 'text',
-      'hidden'                        => false,
-      'options'                       => [],
-      'deleted'                       => false,
-      'displayOrder'                  => 0,
-      'formField'                     => true,
-      'readOnlyValue'                 => false,
-      'readOnlyDefinition'            => false,
-      'mutableDefinitionNotDeletable' => false,
-      'calculated'                    => false,
-      'externalOptions'               => false,
-      'displayMode'                   => 'current_value'
-    } }
-    let(:valid_params) { params.select { |k, _| Hubspot::CompanyProperties::PROPERTY_SPECS[:field_names].include?(k) } }
-
-    describe '.create!' do
-      context 'with no valid parameters' do
-        it 'should return nil' do
-          expect(Hubspot::CompanyProperties.create!({})).to be(nil)
+    describe ".create!" do
+      it "creates a company property" do
+        VCR.use_cassette("company_properties/create_property") do
+          name = "fax_number"
+
+          create_params = {
+            name: name,
+            groupName: "companyinformation",
+          }
+
+          response = Hubspot::CompanyProperties.create!(create_params)
+
+          assert_hubspot_api_request(
+            :post,
+            "/companies/v2/properties",
+            body: {
+              name: name,
+              groupName: "companyinformation",
+              options: []
+            }
+          )
+          expect(response["name"]).to eq(name)
+
+          Hubspot::CompanyProperties.delete!(name)
         end
       end
 
-      context 'with all valid parameters' do
-        cassette 'company_properties/create_property'
+      context "with no valid parameters" do
```

Comment:
Now that our expectation doesn't check the params, should we rename this test?

Reply:
The params are still being checked. `create!` will return early when the params are empty. But to your point, I can update this to be more specific. I'll update this to read `with empty params`.